service/src/tests/integration_tests.rs

```diff
@@ -122,5 +131,8 @@ pub fn shield_funds_workds<E: EnclaveBase + Sidechain>(
 	println!("Sleeping until block with shield funds is finalized...");
 	sleep(Duration::new(10, 0));
 	println!("Syncing light client to look for shield_funds extrinsic");
-	crate::sync_parentchain(enclave_api, &api, last_synced_head)
+
+	let node_api_delegate = NodeApiDelegate::new(api, (*enclave_api).clone());
```

Comment:
It might indeed be cloning here, which gives you the enclave error. What you could try is to change the field to an `Arc<Enclave>` type. `arc.clone()` does not duplicate the object, but does only clone the pointer to the object.

However, according to the above comments, I believe you don't need the `EnclaveApi` anymore.